focus-ios/Blockzilla/Utilities/LocalContentBlocker.swift

```diff
@@ -4,12 +4,111 @@
 
 import Foundation
 import WebKit
+import CryptoKit
 
 class ContentBlockerHelper {
     static let shared = ContentBlockerHelper()
 
     var handler: (([WKContentRuleList]) -> Void)?
 
+    let ruleStore = WKContentRuleListStore.default()
+
+    init() {
+        removeOldListsByHashFromStore { [weak self] in
+            self?.removeOldListsByNameFromStore {
+                self?.getBlockLists { list in
+                    self?.handler?(list)
+                }
+            }
+        }
+    }
+
+    func removeOldListsByNameFromStore(completion: @escaping () -> Void) {
+        var noMatchingIdentifierFoundForRule = false
+
+        ruleStore?.getAvailableContentRuleListIdentifiers { available in
+            guard let available = available else {
+                completion()
+                return
+            }
```

Comment:
@razvanlitianu Is this something we want to log or add a error here 🤔

Reply:
The intent seems to be an early exit kind of "fire and forget".